### DISCUSSION TITLE
Best Practice for Timer Trigger Function in KEDA

### DISCUSSION THREAD

**Discussion by shengdamao** [kedacore/keda#1508] [General]
Is Timer Trigger suitable for KEDA?
If we would like to have one timer trigger function as part of multiple functions, is there any Best Practice Guide? 
 
When function pods scaling up, multiple function host instances will execute Timer Trigger Function duplicatedly. 
Is this an expected behavior? 

If this not expected, what is the best way to aviod Timer Trigger Function being excuted repeated by each function host instance? 
How to gurantee Timer Trigger to be executed only once when using in KEDA?

**Comment by tomkerkhove** [comment]
With function, are you referring to Azure Function? If so, looping in @jeffhollan as this might be good docs for Azure Function + KEDA.

**Reply by shengdamao** [reply]
Yes, I am referring to Azure Function App with Timer Trigger. Thank you for looking into it.

**Reply by tomkerkhove** [reply]
I'll leave that up to @jeffhollan @ahmelsayed but don't think there needs to be any scaling there; unless you want to have scale-to-0?

**Comment by TsuyoshiUshio** [comment]
Hi @shengdamao 
I realize the root cause. Usually timer trigger create a lease lock on a blob container.  It looks reference the hostname on the container. 
However, 
![image](https://user-images.githubusercontent.com/1390976/106943013-b83eec80-66d9-11eb-87e4-8b49f49bb9d0.png)
![image](https://user-images.githubusercontent.com/1390976/106943032-becd6400-66d9-11eb-9953-9e435d79659b.png)

Currently, we need to make it a single container. However, we might have some configuration for k8s. I'll research more. 
If you have any other advice could you give me some? @ahmelsayed

**Reply by TsuyoshiUshio** [reply]
I create an issue to tack the progress. https://github.com/Azure/azure-functions-host/issues/7128

**Reply by TsuyoshiUshio** [reply]
Hi @shengdamao 
I found recommendation approach.  You can add  `AzureFunctionsWebHost__hostid` to your environment variables or local.settings.json (if you use func kubernetes deployment) 

I reproduced the issue and set the environment variables works. 

For example, my local.settings.json is 

```json
{
    "IsEncrypted": false,
    "Values": {
        "AzureWebJobsStorage": "MY_CONNECTION_STRING",
        "FUNCTIONS_WORKER_RUNTIME": "dotnet",
        "AzureFunctionsWebHost__hostid": "singletonTimer"
    }
}
```
https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=csharp#function-apps-sharing-storage
